### DISCUSSION TITLE
Best practice for refreshing the whole vector collections after a new model is trained

### DISCUSSION THREAD

**Discussion by Siu-Keung** [milvus-io/milvus#22585] [Q&A and General discussion]
Hi all, I am new to Milvus and want to learn the best practice to refresh the **whole** vector collections when our model is **re-trained**.

Imagine the below production scenario:
1. We train a NLP model which can transform the raw texts to vectors, named **Model A**.
2. We then use this model to insert some data into Milvus DB and handle user's queries **simultaneously**.
3. We now train a **new** model named **Model B**, which has better performance than Model A. **Thus we want to put this new model in the production env to replace Model A**.

I would like to collect advices here on how we can execute the **step 3  gracefully**? Since once we decide to replace Model A with Model B, then all of the existing vectors stored in the  DB need to be re-generated via Model B. This is **time-consuming** and user might get the **intermediate** result if he/she execute querying during this period. Do we have a mechanism to ensure this kind of full collection update is transparent to the users? Thanks in advance.

**Comment by jiaoew1991** [comment]
Hi, you can use the collection alias feature at https://milvus.io/docs/collection_alias.md. Expose the collection alias to users and switch it to the new table after rewriting data in the new table.
